modules/gapi/include/opencv2/gapi/core.hpp

```diff
@@ -508,6 +535,83 @@ namespace core {
             return in.withType(in.depth, in.chan).withSize(dsize);
         }
     };
+
+    G_TYPED_KERNEL(
+        GKMeansND,
+        <std::tuple<GOpaque<double>,GMat,GMat>(GMat,int,TermCriteria,int,KmeansFlags,GMat)>,
+        "org.opencv.core.kmeansND") {
+
+        static std::tuple<GOpaqueDesc,GMatDesc,GMatDesc>
+        outMeta(const GMatDesc& in, int K, const TermCriteria&, int, KmeansFlags flags,
+                const GMatDesc& labels) {
+            int elemSize = 0, quantity = 0;
+            std::tie(quantity, elemSize) = checkVector(in.chan, in.size, in.depth, CV_32F);
+            GMatDesc out_labels, centers;
+            if (flags & KMEANS_USE_INITIAL_LABELS)
+            {
+                GAPI_Assert(labels.depth == CV_32S && labels.chan == 1 &&
+                            labels.size.area() == quantity * 1 &&
```

Comment:
` * 1`? Raises more questions than answers.

Reply:
This check looks like part of `isPointsVector` for first look.

Reply:
I meant `area() = width * height` should be 1 x quantity :)
Anyway, refactored to use `checkVector`